google-cloud-bigquery/src/test/java/com/google/cloud/bigquery/it/ITBigQueryTest.java

```diff
@@ -2589,6 +2589,68 @@ public void testCopyJob() throws InterruptedException, TimeoutException {
+    TableInfo tableInfo = TableInfo.of(sourceTableId, tableDefinition);
+    Table createdTable = bigquery.create(tableInfo);
+    assertNotNull(createdTable);
+
+    // Create snapshot table using source table as the base table
+    TableId snapshotTableId = TableId.of(DATASET, snapshotTableName);
+    CopyJobConfiguration snapshotConfiguration =
+        CopyJobConfiguration.newBuilder(snapshotTableId, sourceTableId)
+            .setOperationType("SNAPSHOT")
+            .build();
+    Job createdJob = bigquery.create(JobInfo.of(snapshotConfiguration));
+    CopyJobConfiguration createdConfiguration = createdJob.getConfiguration();
+    assertNotNull(createdConfiguration.getSourceTables());
+    assertNotNull(createdConfiguration.getOperationType());
+    assertNotNull(createdConfiguration.getDestinationTable());
+    Job completedJob = createdJob.waitFor();
+    assertNull(completedJob.getStatus().getError());
+    Table snapshotTable = bigquery.getTable(DATASET, snapshotTableName);
+    assertNotNull(snapshotTable);
+    assertEquals(snapshotTableId.getDataset(), snapshotTable.getTableId().getDataset());
+    assertEquals(snapshotTableName, snapshotTable.getTableId().getTable());
+    assertEquals(TABLE_SCHEMA, snapshotTable.getDefinition().getSchema());
+
+    // Update snapshot table with some description
+    Table updatedSnapshotTable =
+        bigquery.update(
+            snapshotTable.toBuilder().setDescription("This is a snapshot table").build());
+    assertEquals(updatedSnapshotTable.getDescription(), "This is a snapshot table");
+    assertEquals(snapshotTableName, snapshotTable.getTableId().getTable());
+
+    // Restore base table to a new table
+    String restoredTableName = "test_restore_table";
+    TableId restoredTableId = TableId.of(DATASET, restoredTableName);
+    CopyJobConfiguration restoreConfiguration =
+        CopyJobConfiguration.newBuilder(restoredTableId, snapshotTableId)
+            .setOperationType("RESTORE")
+            .build();
+    Job createdRestoreJob = bigquery.create(JobInfo.of(restoreConfiguration));
+    CopyJobConfiguration createdRestoreConfiguration = createdRestoreJob.getConfiguration();
+    assertNotNull(createdRestoreConfiguration.getSourceTables());
+    assertNotNull(createdRestoreConfiguration.getOperationType());
+    assertNotNull(createdRestoreConfiguration.getDestinationTable());
+    Job completedRestoreJob = createdRestoreJob.waitFor();
+    assertNull(completedRestoreJob.getStatus().getError());
+    Table restoredTable = bigquery.getTable(DATASET, restoredTableName);
+    assertNotNull(restoredTable);
+    assertEquals(restoredTableId.getDataset(), restoredTable.getTableId().getDataset());
+    assertEquals(restoredTableName, restoredTable.getTableId().getTable());
+    assertEquals(TABLE_SCHEMA, restoredTable.getDefinition().getSchema());
+
+    // Clean up
+    assertTrue(createdTable.delete());
```

Comment:
You're asserting cleanup of the origin table and the snapshot, but not restoredTable.

Reply:
Added cleanup for `restoredTable`.